packages/@lwc/engine/src/framework/upgrade.ts

```diff
@@ -110,13 +48,14 @@ export function createElement(sel: string, options: CreateElementOptions): HTMLE
     }
 
     const mode = options.mode !== 'closed' ? 'open' : 'closed';
+    registerTagName(sel);
 
     // Create element with correct tagName
     const element = document.createElement(sel);
-    if (!isUndefined(getAssociatedVMIfPresent(element))) {
-        // There is a possibility that a custom element is registered under tagName,
-        // in which case, the initialization is already carry on, and there is nothing else
-        // to do here.
+    if (!isUpgradableElement(element)) {
```

Comment:
No need to check if the created element is upgradable at this point. Line 51, the `registerTagName` returns `false` if the element name is already used.

Reply:
no, imagine this gets called twice! the first time, it register it, the second time, it doesn't but still will be upgradable. same applies for multi-engines in the future.